src/Centreon/Application/Controller/Monitoring/CommentController.php

```diff
@@ -0,0 +1,306 @@
+        */
+        $contact = $this->getUser();
+        $this->commentService->filterByContact($contact);
+
+       /*
+        * Validate the content of the request against the JSON schema validator
+        */
+        $receivedData = $this->validateAndRetrievePostData(
+            $request,
+            'config/json_validator/latest/Centreon/Comment/CommentResources.json'
+        );
+
+        /**
+         * If user has no rights to add a comment for host and/or service
+         * return view with unauthorized HTTP header response
+         */
+        if (!$this->hasCommentRightsForResources($contact, $receivedData['resources'])) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+
+        /**
+         * Dissect the results to extract the hostids and serviceids from it
+         */
+        $resourceIds = [];
+        $comments = [];
+
+        $now = new \DateTime();
+
+        foreach ($receivedData['resources'] as $commentResource) {
+            $date = ($commentResource['date'] !== null) ? new \DateTime($commentResource['date']) : $now;
+            $comments[$commentResource['id']] = (new Comment($commentResource['id'], $commentResource['comment']))
+                ->setDate($date)
+                ->setParentResourceId($commentResource['parent']['id']);
+
+            if ($commentResource['type'] === ResourceEntity::TYPE_HOST) {
+                $resourceIds['host'][] = $commentResource['id'];
+            } elseif ($commentResource['type'] === ResourceEntity::TYPE_SERVICE) {
+                $resourceIds['service'][] = [
+                    'host_id' => $commentResource['parent']['id'],
+                    'service_id' => $commentResource['id']
+                ];
+            }
+        }
+
+        $this->commentService->addResourcesComment($comments, $resourceIds);
+
+        return $this->view(null, Response::HTTP_NO_CONTENT);
+    }
+
+    /**
+     * Entry point to add a comment on a host resource
+     *
+     * @param Request $request
+     * @param int $hostId ID of the host
+     * @return View
+     * @throws \Exception
+     * @throws \InvalidArgumentException
+     */
+    public function addHostComment(
+        Request $request,
+        int $hostId
+    ): View {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+        /**
+         * @var Contact $contact
+         */
+        $contact = $this->getUser();
+        $this->commentService->filterByContact($contact);
+
+        /**
+         * Checking that user is allowed to add a comment for a host resource
+         */
+        if (!$contact->isAdmin() && !$contact->hasRole(Contact::ROLE_HOST_ADD_COMMENT)) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+
+        $receivedData = $this->validateAndRetrievePostData(
+            $request,
+            'config/json_validator/latest/Centreon/Comment/Comment.json'
+        );
+
+        if (!empty($receivedData)) {
```

Comment:
When is the data received empty?
Doesn't the json validator verify that valid data is received?